### PR TITLE
MAGECLOUD-2820: Implement isolated connections mechanism

### PR DESCRIPTION
**Pull request by arhiopterecs**
### Description
Fixed the database query engine.
Suppress PDO warnings to work around the bug https://bugs.php.net/bug.php?id=63812, https://bugs.php.net/bug.php?id=74401
### Fixed Issues (if relevant)
[MAGECLOUD-2820](https://magento2.atlassian.net/browse/MAGECLOUD-2820)

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by andriyShevtsov** [comment]
QA approved

File: patches.json [hunk 1]
```diff
@@ -138,6 +138,12 @@
     "Add the possibility to install Magento without admin creation" : {
       "2.1.4 - 2.2.1": "MAGECLOUD-2573__installation_without_admin_creation__2.1.4.patch",
       "2.2.2 - 2.2.7": "MAGECLOUD-2573__installation_without_admin_creation__2.2.2.patch"
+    },
+    "Suppress PDO warnings to work around PHP bugs #63812, #74401": {
+      "2.1.4": "MAGECLOUD-2820__implement_isolated_connections_mechanism__2.1.4.patch",
+      "2.1.5 - 2.1.12": "MAGECLOUD-2820__implement_isolated_connections_mechanism__2.1.5.patch",
+      "2.1.13 - 2.1.17": "MAGECLOUD-2820__implement_isolated_connections_mechanism__2.1.13.patch",
+      "2.2.0 - 2.2.8 || 2.3.0 - 2.3.1": "MAGECLOUD-2820__implement_isolated_connections_mechanism__2.2.0.patch"
```
Comment (NadiyaS) [inline patches.json]: Wasn't patch delivered to 2.3.x?

File: patches/MAGECLOUD-2820__implement_isolated_connections_mechanism__2.1.13.patch [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+  *
+  * @author      Magento Core Team <core@magentocommerce.com>
+  */
+-namespace Magento\Framework\DB\Statement\Pdo;
+-
+-use Magento\Framework\DB\Statement\Parameter;
+-
+ class Mysql extends \Zend_Db_Statement_Pdo
+ {
++
```
Comment (NadiyaS) [inline patches/MAGECLOUD-2820__implement_isolated_connections_mechanism__2.1.13.patch:218]: Have you created the patch from fix to core or have you created it after passing the tests?
If so, you should not have this line there.

**Review by NadiyaS — APPROVED** [review]

File: patches/MAGECLOUD-2820__implement_isolated_connections_mechanism__2.1.13.patch (hunk 2 above)
Comment (arhiopterecs) [reply patches/MAGECLOUD-2820__implement_isolated_connections_mechanism__2.1.13.patch:218]: @NadiyaS, I created the patch from fix to core, i did it after passing the tests

**Pull request merged**